test/k8s-integration/main.go

```diff
@@ -436,10 +439,20 @@ func handle() error {
 
 	switch testParams.deploymentStrategy {
 	case "gke":
-		testParams.cloudProviderArgs, err = getGKEKubeTestArgs(*gceZone, *gceRegion, testParams.imageType)
-		if err != nil {
+		// Build params for both kubetest and kubetest2 for debugging.
+		testParams.cloudProviderArgs, err = getGKEKubeTestArgs(*gceZone, *gceRegion, testParams.imageType, false /* useKubeTest2 */)
+		if err != nil && !*useKubeTest2 {
 			return fmt.Errorf("failed to build GKE kubetest args: %v", err)
 		}
+		testParams.cloudProviderArgsK2, err = getGKEKubeTestArgs(*gceZone, *gceRegion, testParams.imageType, true /* useKubeTest2 */)
+		if err != nil && *useKubeTest2 {
+			return fmt.Errorf("failed to build GKE kubetest2 args: %v", err)
+		}
+	case "gce":
+		testParams.cloudProviderArgsK2 = []string{
+			"--legacy-mode",
```

Comment:
could you add a comment for what legacy-mode does? I always forget, and since now kubetest1 can be considered a legacy mode it would be good to not be ambiguous.

Reply:
Good call. The naming isn't very helpful.